typescript/src/main.ts

```diff
@@ -95,6 +134,21 @@ function appendMetalTableRow(metal: metals.Metal, table: HTMLTableElement) {
   resultCell.id = "result_" + metal.idSuffix;
 }
 
+function updateRow(metal: metals.Metal) {
+  const id = metal.idSuffix;
+  (<HTMLInputElement>document.getElementById("affinity_" + id)).value =
```

Comment:
Are all these casts necessary? (Realise they were already in the `reset` method so apologies if I added them originally, just noticed them now the code has moved!)

Reply:
Sorry, the input element casts are probably necessary - I missed that they were inputs when reviewing yesterday. But can you try without the HTMLTableCellElement cast? I think innerText should work on HTMLElement.

Reply:
I think that's what I tried initially, I'll let you know if tit works when I get to this part.